### PR TITLE
docs: Document how to turn on strict mode for an existing project. 

### PR DESCRIPTION
**Pull request by chrisjsimpson**
Document how to turn on strict mode for an existing project. 

It's just as, if not more likely, developers will want to turn this on for an existing project so it's useful to document how to do that.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

Issue Number: N/A

### CODE REVIEW

File: aio/content/guide/strict-mode.md [hunk 1]
```diff
@@ -28,3 +28,40 @@ To create a new application in the strict mode within an existing non-strict wor
 ng generate application [project-name] --strict
 
 </code-example>
+
+#### To enable strict mode for an *existing project*
```
Comment (kapunahelewong) [inline aio/content/guide/strict-mode.md:32]:
```suggestion
#### Enabling strict mode for an existing project
```

File: aio/content/guide/strict-mode.md [hunk 2]
```diff
@@ -28,3 +28,40 @@ To create a new application in the strict mode within an existing non-strict wor
...
+
+Edit your `angular.json` and add `@schematics/angular:application` with `"strict": true` to the `schematics` property:
```
Comment (kapunahelewong) [inline aio/content/guide/strict-mode.md:34]:
```suggestion
Edit your `angular.json` by adding `@schematics/angular:application` with `"strict": true` to the `schematics` property as follows.
```

File: aio/content/guide/strict-mode.md [hunk 3]
```diff
@@ -28,3 +28,40 @@ To create a new application in the strict mode within an existing non-strict wor
...
+      "projectType": "application",
+      "schematics": {
+        "@schematics/angular:application": {
+          "strict": true
+        }
+      }
+   ...
+ ```
+
+Edit `compilerOptions` in `tsconfig.json` to include:
```
Comment (kapunahelewong) [inline aio/content/guide/strict-mode.md:49]:
```suggestion
Add the following to `compilerOptions` in `tsconfig.json`.
```

File: aio/content/guide/strict-mode.md [hunk 4]
```diff
@@ -28,3 +28,40 @@ To create a new application in the strict mode within an existing non-strict wor
...
+
+```
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+```
```
Comment (kapunahelewong) [inline aio/content/guide/strict-mode.md:56]:
```suggestion
<code-example header="tsconfig.json">
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
</code-example>
```

File: aio/content/guide/strict-mode.md [hunk 5]
```diff
@@ -28,3 +28,40 @@ To create a new application in the strict mode within an existing non-strict wor
...
+  "projects": {
+    "app": {
+      "projectType": "application",
+      "schematics": {
+        "@schematics/angular:application": {
+          "strict": true
+        }
+      }
+   ...
+ ```
```
Comment (kapunahelewong) [inline aio/content/guide/strict-mode.md:47]:
```suggestion
<code-example header="angular.json">
  ...
  "projects": {
    "app": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:application": {
          "strict": true
        }
      }
   ...
</code-example>
```

File: aio/content/guide/strict-mode.md [hunk 6]
```diff
@@ -28,3 +28,40 @@ To create a new application in the strict mode within an existing non-strict wor
...
+```
+
+Add a new property `angularCompilerOptions` to `tsconfig.json`:
+```
+  "angularCompilerOptions": {
+    "strictInjectionParameters": true,
+    "strictInputAccessModifiers": true,
+    "strictTemplates": true
+  }
+```
```
Comment (kapunahelewong) [inline aio/content/guide/strict-mode.md:65]:
```suggestion
<code-example header="tsconfig.json">
  "angularCompilerOptions": {
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
</code-example>
```

File: aio/content/guide/strict-mode.md [hunk 7]
```diff
@@ -28,3 +28,40 @@ To create a new application in the strict mode within an existing non-strict wor
...
+
+Finally add `"no-any": true` to `rules` in `tslint.json`.
```
Comment (kapunahelewong) [inline aio/content/guide/strict-mode.md:67]:
```suggestion
Finally, add `"no-any": true` to `rules` in `tslint.json`.
```

**Review by kapunahelewong — COMMENTED** [review]
This is a really nice addition, @chrisjsimpson. Thank you. I've added some docs style-related comments, but we'll still need engineering review.

We try to use code snippets that are part of examples, but I believe json is an exception. So,  in the cases where we do have in-line hardcoded examples, we've been using `<code-example>` rather than code fencing so we can have a header on it. The header includes the file that the person is working in, which helps orient the reader. cc @gkalpak 

Feel free to tag me with any questions.

File: aio/content/guide/strict-mode.md (hunk 7 above)
Comment (IgorMinar) [inline aio/content/guide/strict-mode.md:67]: `no-any` is no longer part of the strict mode as of v11. Can you please remove this part?

Comment (IgorMinar) [inline aio/content/guide/strict-mode.md:67]: I don't see a mention of stricter bundle budgets, can you please add a mention of updating those?

**Review by IgorMinar — CHANGES_REQUESTED** [review]
overall this looks great @chrisjsimpson! can you please incorporate the review feedback as a fixup commit and rerequest a review? thank you!

**Comment by petebacondarwin** [comment]
@chrisjsimpson - could you address the outstanding comments and squash all the commits into one? Thanks!

**Comment by AndrewKushnir** [comment]
@alan-agius4 could you please have a quick look at this PR when you get a chance (to double check whether the content is correct given the strict mode changes in CLI)?

**Comment by alan-agius4** [comment]
@AndrewKushnir, it looks like this PR is stalled. @IgorMinar's comments are still not addressed.

**Comment by AndrewKushnir** [comment]
Closing this PR due to the lack of reply. Please reopen if needed.

**Pull request closed without merging**